src/Controls/samples/Controls.Sample/Pages/Controls/CollectionViewGalleries/HeaderFooterGalleries/HeaderFooterEmptyView.xaml.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.Maui.Controls;
+using Microsoft.Maui.Controls.Xaml;
+
+namespace Maui.Controls.Sample.Pages.CollectionViewGalleries.HeaderFooterGalleries
+{
+	[XamlCompilation(XamlCompilationOptions.Compile)]
+	public partial class HeaderFooterEmptyView : ContentPage
+	{
+		public HeaderFooterEmptyView()
+		{
+			InitializeComponent();
+			UpdateEmptyView();
+		}
+
+		LayoutOptions _horizontalOptions = LayoutOptions.Center;
+		LayoutOptions _verticalOptions = LayoutOptions.Center;
+
+		void SetHorizontalOptions(object sender, CheckedChangedEventArgs _)
+		{
+			var radioButton = (RadioButton)sender;
+			if (radioButton.IsChecked)
+			{
+				_horizontalOptions = (LayoutOptions)radioButton.Value;
+				UpdateEmptyView();
+			}
+		}
+
+		void SetVerticalOptions(object sender, CheckedChangedEventArgs _)
+		{
+			var radioButton = (RadioButton)sender;
+			if (radioButton.IsChecked)
+			{
+				_verticalOptions = (LayoutOptions)radioButton.Value;
+				UpdateEmptyView();
+			}
+		}
+
+		void UpdateEmptyView()
+		{
+			Collection.EmptyView = new Label
+			{
+				Text = "Nothing to show",
+				HorizontalOptions = _horizontalOptions,
+				VerticalOptions = _verticalOptions
+			};
```

Comment:
Not sure re-creation is the best option as it may involve complex things. @PureWeen not sure the original plan here, but should the empty view always fill the entire control? Assuming centered text was desird, would this involve some layout and then a label, or should a label just work.

Not sure if there is an expected/designed functionality for how backgrounds and things work.

Reply:
With regards to filling the entire control there is different behavior across the platforms. On iOS it does. But not on Android and Windows.

Personally I would like it to fill, and let it be up to the UI designer to align it as they see fit. For example, an entire page (with a single CollectionView) can look quite meek if there is just an empty view message on the very top.

Reply:
> should the empty view always fill the entire control

Yes. The fact that it's not doing that on some platforms is a bug that needs to be addressed.

Reply:
A note on the emptyview not listening to changes on HorizontalOptions / VerticalOptions. It takes an effort to setup event listeners and dealing with teardown and such properly. If that is a wanted behavior I think it should be dealt with in a separate PR. It is strictly an exisiting case and not something introduced with this PR.